File: Dihya/scripts/onboarding_backend.js
// Backend Node.js – Onboarding utilisateur sécurisé, RGPD, audit, fallback
const express = require('express');
const app = express();
app.use(express.json());

app.post('/api/onboarding/', (req, res) => {
  const { email, lang = 'fr' } = req.body;
  if (!email) return res.status(400).json({ error: 'Email requis', lang });
  // RGPD: log anonymisé
  console.log(`[ONBOARDING] ${new Date().toISOString()} - ${email.slice(0,2)}***@*** - ${lang}`);
  // Fallback IA open source simulé
  if (process.env.FALLBACK_IA === '1') {
    return res.json({ message: 'Réponse fallback IA', lang, fallback: true });
  }
  res.json({ message: `Bienvenue ${email}!`, lang, fallback: false });
});

app.listen(8002, () => console.log('Node backend running on 8002'));

File: Dihya/scripts/onboarding_backend.py
# Backend Python (Flask) – Onboarding utilisateur sécurisé, multilingue, RGPD, audit, fallback
from flask import Flask, request, jsonify
import logging, os
from datetime import datetime

app = Flask(__name__)
logging.basicConfig(level=logging.INFO)

@app.route('/api/onboarding/', methods=['POST'])
def onboarding():
    data = request.json or {}
    email = data.get('email')
    lang = data.get('lang', 'fr')
    if not email:
        return jsonify({'error': 'Email requis', 'lang': lang}), 400
    # RGPD: log anonymisé
    logging.info(f"[ONBOARDING] {datetime.utcnow()} - {email[:2]}***@*** - {lang}")
    # Fallback IA open source simulé
    if os.environ.get('FALLBACK_IA', '0') == '1':
        return jsonify({'message': 'Réponse fallback IA', 'lang': lang, 'fallback': True})
    return jsonify({'message': f'Bienvenue {email}!', 'lang': lang, 'fallback': False})

if __name__ == '__main__':
    app.run(port=8001)
